tests/exploits/cve_2024_8956_test.py

```diff
@@ -0,0 +1,93 @@
+"""Unit tests for VHD PTZ Camera Authentication Bypass: CVE-2024-8956"""
+
+import requests_mock as req_mock
+from requests import exceptions as requests_exceptions
+
+from agent import definitions
+from agent.exploits import cve_2024_8956
+
+
+def testVHDPTZAuthBypass_whenVulnerable_reportFinding(
+    requests_mock: req_mock.mocker.Mocker,
+) -> None:
+    """Test case: when target is vulnerable (firmware 6.3.32)."""
+    requests_mock.get(
+        "http://localhost:80/cgi-bin/param.cgi?get_device_conf",
+        text='devname="ptzoptics" devtype="VX630A" versioninfo="SOC v6.3.32 - ARM 6.3.51THI" serial_num="r1j04260027" device_model="F53.HI"',
+        status_code=200,
+    )
+
+    exploit_instance = cve_2024_8956.VHDPTZAuthBypassExploit()
+    target = definitions.Target("http", "localhost", 80)
+
+    vulnerabilities = exploit_instance.check(target)
+
+    assert len(vulnerabilities) == 1
+
+    vulnerability = next(
+        v
+        for v in vulnerabilities
+        if v.entry.title == "ValueHD PTZ Camera Authentication Bypass"
+    )
+    assert vulnerability.entry.references == {
+        "nvd.nist.gov": "https://nvd.nist.gov/vuln/detail/CVE-2024-8956"
+    }
```

Comment:
please apply to the rest: https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/

```suggestion
def testVHDPTZAuthBypass_whenVulnerable_reportFinding(
    requests_mock: req_mock.mocker.Mocker,
) -> None:
    """Test case: when target is vulnerable (firmware 6.3.32)."""
    requests_mock.get(
        "http://localhost:80/cgi-bin/param.cgi?get_device_conf",
        text='devname="ptzoptics" devtype="VX630A" versioninfo="SOC v6.3.32 - ARM 6.3.51THI" serial_num="r1j04260027" device_model="F53.HI"',
        status_code=200,
    )
    exploit_instance = cve_2024_8956.VHDPTZAuthBypassExploit()
    target = definitions.Target("http", "localhost", 80)

    vulnerabilities = exploit_instance.check(target)

    assert len(vulnerabilities) == 1
    vulnerability = next(
        v
        for v in vulnerabilities
        if v.entry.title == "ValueHD PTZ Camera Authentication Bypass"
    )
    assert vulnerability.entry.references == {
        "nvd.nist.gov": "https://nvd.nist.gov/vuln/detail/CVE-2024-8956"
    }
```